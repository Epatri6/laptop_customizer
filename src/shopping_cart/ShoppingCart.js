import React from 'react';
import './ShoppingCart.css'
import ShoppingSummary from '../shopping_summary/ShoppingSummary';
import ShoppingTotal from '../shopping_total/ShoppingTotal';

export default class ShoppingCart extends React.Component {

  render() {
    return (
      <section className="main__summary">
      <h2>Your cart</h2>
      <ShoppingSummary selectedFeatures={this.props.selectedFeatures} />
      <ShoppingTotal selectedFeatures={this.props.selectedFeatures} />
      </section>
    );
  }

  static defaultProps = {
    selectedFeatures: {},
  }
}