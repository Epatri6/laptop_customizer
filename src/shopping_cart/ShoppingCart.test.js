import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingCart from './ShoppingCart';
import renderer from 'react-test-renderer';

describe('Testing ShoppingCart Component', () => {

  const testProps = {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    'Operating System': {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    }
  };

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingCart selectedFeatures={testProps}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingCart />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<ShoppingCart selectedFeatures={testProps}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<ShoppingCart />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});