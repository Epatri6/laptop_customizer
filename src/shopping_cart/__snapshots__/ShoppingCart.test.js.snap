// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingCart Component renders as expected 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <div>
    <div
      className="summary__option"
    >
      <div
        className="summary__option__label"
      >
        Processor
         
      </div>
      <div
        className="summary__option__value"
      >
        17th Generation Intel Core HB (7 Core with donut spare)
      </div>
      <div
        className="summary__option__cost"
      >
        $700.00
      </div>
    </div>
    <div
      className="summary__option"
    >
      <div
        className="summary__option__label"
      >
        Operating System
         
      </div>
      <div
        className="summary__option__value"
      >
        Ubuntu Linux 16.04
      </div>
      <div
        className="summary__option__cost"
      >
        $200.00
      </div>
    </div>
  </div>
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Total
    </div>
    <div
      className="summary__total__value"
    >
      $900.00
    </div>
  </div>
</section>
`;

exports[`Testing ShoppingCart Component renders as expected without props 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <div />
  <div
    className="summary__total"
  >
    <div
      className="summary__total__label"
    >
      Total
    </div>
    <div
      className="summary__total__value"
    >
      $0.00
    </div>
  </div>
</section>
`;
